packages/teleport/src/components/FilterByLabelList/FilterByLabelList.tsx

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright 2021 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useMemo } from 'react';
+import { useLocation } from 'react-router';
+import { Option } from 'shared/components/Select';
+import SelectFilter from 'teleport/components/SelectFilter';
+import history from 'teleport/services/history';
+
+export default function FilterableList({
+  data = [],
+  TableComponent,
+  ...tableProps
+}: Props) {
+  const { search, pathname } = useLocation();
+  const labelsFromQuery = useMemo<Option[]>(() => {
+    const searchParams = new URLSearchParams(search);
+    const query = searchParams.get('labels');
+    if (!query || query === 'null') {
+      return [];
+    }
+
+    return decodeURIComponent(query)
+      .split(',')
+      .map(label => ({ value: label, label }));
+  }, []);
```

Comment:
Shouldn't it depend on `search`? Currently, the UI won't react on manual URL change.

Reply:
no, it wasn't intended to change per change in URL, this should be combined with `selectedFilter`